axis.go

```diff
@@ -377,17 +378,22 @@ func (DefaultTicks) Ticks(min, max float64) (ticks []Tick) {
 	}
 	majorDelta := float64(majorMult) * tens
 	val := math.Floor(min/majorDelta) * majorDelta
-	prec := precisionOf(majorDelta)
 	for val <= max {
 		if val >= min && val <= max {
-			ticks = append(ticks, Tick{Value: val, Label: formatFloatTick(val, prec)})
-		}
-		if math.Nextafter(val, val+majorDelta) == val {
-			break
+			//Makes a list of values.
+			labels = append(labels, val)
 		}
 		val += majorDelta
 	}
 
+	//Makes a list of labels with a level of precision where all the labels are different.
+	newLabels = adjustPrecision(labels)
+
+	for j := 0; j < len(labels); j++ {
+		//Makes a list of big ticks.
+		ticks = append(ticks, Tick{Value: labels[j], Label: formatFloatTick(newLabels[j], -1)})
```

Comment:
`ticks = append(ticks, Tick{Value: v, Label: formatFloatTick(newLabels[j], -1)})`

Also, I suspect that if you get `adjustPrecision` to return the precision that was decided on as a second return parameter, if you pass that to `formatFloatTick` you would get rid of the problem with the very high exponent test failure (also below).

Reply:
It did not work, as formatFloatTick's precision is different, as it works with strings.

Reply:
Yes, thanks, that's true. This highlights something that was at the back of my mind. When you and Seb were discussing your approach previously I had thought that you were going to do rendering and assess whether the rendered labels differed. I had forgotten this. It looks like there is an impedance mismatch in the approach that is being used that that approach would not have. We can forge ahead with this approach, since the basic structure is the same, but I think this particular problem goes away with that approach, the n is smaller (though the constants are larger).